File: Upnp/Public/Cs/TestPerformanceDv.cs
using System;
using OpenHome.Net.Core;
using OpenHome.Net.Device;

namespace OpenHome.Net.Device
{
    class Program
    {
        public static void Main(string[] args)
        {
            Console.WriteLine("TestPerformanceDvCs - starting ('q' to quit)");
            InitParams initParams = new InitParams();
            using (Library lib = Library.Create(initParams))
            {
                lib.StartDv();
                DeviceBasic device = new DeviceBasic();
                while (Console.ReadKey(true).KeyChar != 'q') ;
                device.Dispose();
            }
        }
    }
}

File: OpenHome/Net/Bindings/Java/org/openhome/net/device/tests/TestPerformanceDv.java
package org.openhome.net.device.tests;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

import org.openhome.net.core.InitParams;
import org.openhome.net.core.Library;

public class TestPerformanceDv {
	
	public static void main(String[] args) throws IOException
    {
        System.out.println("TestPerformanceDvJava - starting ('q' to quit)");
        InitParams initParams = new InitParams();
        Library lib = Library.create(initParams);
        lib.startDv();
        DeviceBasic device = new DeviceBasic();
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
        try
        {
        	while(in.read() != 'q') ;
        }
        catch (IOException io)
        {
        	System.err.println("ERROR: Exiting abnormally");
        }
        device.dispose();
    }
}
